ent/encryption/encryption.cc

```diff
@@ -823,6 +823,30 @@ class encryption_file_io_extension : public sstables::file_io_extension {
         });
         co_return sink;
     }
+
+    future<data_source> wrap_source(const sstables::sstable& sst, sstables::component_type type, data_source source) override {
+        switch (type) {
+        case sstables::component_type::Scylla:
+        case sstables::component_type::TemporaryTOC:
+        case sstables::component_type::TOC:
+            co_return source;
+        case sstables::component_type::CompressionInfo:
+        case sstables::component_type::CRC:
+        case sstables::component_type::Data:
+        case sstables::component_type::Digest:
+        case sstables::component_type::Filter:
+        case sstables::component_type::Index:
+        case sstables::component_type::Statistics:
+        case sstables::component_type::Summary:
+        case sstables::component_type::TemporaryStatistics:
+        case sstables::component_type::Unknown:
```

Comment:
Yikes, why not just "default:" as `wrap_sink` does?

Reply:
This is the way!

https://github.com/scylladb/scylladb/pull/23695#discussion_r2041864711